rs/index/src/ivf/writer.rs

```diff
@@ -0,0 +1,322 @@
+        let mut data_buffer_writer = BufWriter::new(&mut data_file);
+
+        // Write data
+        for vector in ivf.dataset.iter() {
+            for elem in vector.iter() {
+                let bytes_written = wrap_write(&mut data_buffer_writer, &elem.to_le_bytes())?;
+                if bytes_written != 4 {
+                    return Err(anyhow!(
+                        "Expected to write 4 bytes, but wrote {}",
+                        bytes_written,
+                    ));
+                }
+            }
+        }
+
+        // Centroids writer
+        let centroids_path = format!("{}/centroids", self.base_directory);
+        let mut centroids_file = File::create(centroids_path).unwrap();
+        let mut centroids_buffer_writer = BufWriter::new(&mut centroids_file);
+
+        // Write centroids
+        for centroid in ivf.centroids.iter() {
+            for elem in centroid.iter() {
+                let bytes_written = wrap_write(&mut centroids_buffer_writer, &elem.to_le_bytes())?;
+                if bytes_written != 4 {
+                    return Err(anyhow!(
+                        "Expected to write 4 bytes, but wrote {}",
+                        bytes_written,
+                    ));
+                }
+            }
+        }
+
+        // Inverted lists writer
+        let inverted_lists_path = format!("{}/inverted_lists", self.base_directory);
+        let mut inverted_lists_file = File::create(inverted_lists_path).unwrap();
+        let mut inverted_lists_buffer_writer = BufWriter::new(&mut inverted_lists_file);
+
+        // Write interted lists
+        for (centroid_idx, vector_indices) in &ivf.inverted_lists {
+            let ci = *centroid_idx as u32;
+            wrap_write(&mut inverted_lists_buffer_writer, &ci.to_le_bytes())?;
+            // Write the number of vectors in this centroid.
+            let vectors_in_centroid = vector_indices.len() as u64;
+            wrap_write(
+                &mut inverted_lists_buffer_writer,
+                &vectors_in_centroid.to_le_bytes(),
+            )?;
+            for vector_idx in vector_indices.iter() {
+                let vi = *vector_idx as u64;
+                wrap_write(&mut inverted_lists_buffer_writer, &vi.to_le_bytes())?;
+            }
+        }
+
+        data_buffer_writer.flush().unwrap();
+        centroids_buffer_writer.flush().unwrap();
+        inverted_lists_buffer_writer.flush().unwrap();
+
+        self.combine_files(header)?;
+
+        // Remove edges, points, edge_offsets, level_offsets. Ok to ignore errors
+        fs::remove_file(format!("{}/data", self.base_directory)).unwrap_or_default();
+        fs::remove_file(format!("{}/centroids", self.base_directory)).unwrap_or_default();
+        fs::remove_file(format!("{}/inverted_lists", self.base_directory)).unwrap_or_default();
+
+        Ok(())
+    }
+
+    /// Read file and append to the writer
+    fn append_file_to_writer(
```

Comment:
can you move this to utils? I think it's also used in the hnsw writer. move it to `rs/utils/io.rs`